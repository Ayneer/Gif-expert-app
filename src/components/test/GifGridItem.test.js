import { shallow } from 'enzyme';
import { GifGridItem } from '../GifGridItem';

describe('Gif Grid Item suit test', () => {

    test('should show the component successfully', () => {
        const title = 'My test title';
        const url = 'http//test.com';

        const wrapper = shallow(<GifGridItem title={title} url={url} />);

        const htmlImgSrc = wrapper.find('img').prop('src');
        const htmlImgAlt = wrapper.find('img').prop('alt');
        const htmlPText = wrapper.find('p').text();
        const htmlDivClass = wrapper.find('div').prop('className');

        expect(wrapper).toMatchSnapshot();
        expect(htmlImgSrc).toBe(url);
        expect(htmlImgAlt).toBe(title);
        expect(htmlPText).toBe(title);
        expect(htmlDivClass.includes('animate__zoomIn')).toBeTruthy();
    });


});
