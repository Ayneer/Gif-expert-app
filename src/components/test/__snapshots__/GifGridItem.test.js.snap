// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid Item suit test should show the component successfully 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="My test title"
    src="http//test.com"
  />
  <p>
    My test title
  </p>
</div>
`;
